pkg/resourcemanager/controller/secret/reconciler.go

```diff
@@ -66,6 +68,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels[v1alpha1.ReferencedBy]; !ok {
```

Comment:
What happens when Secret is no longer referenced? We cannot remove the label when Secret is no longer referenced because this would break the change we want to introduce in the helper pkg? Then, let's document properly that this label is add only and not removed when the Secret is no longer referenced.

Reply:
Should we add logic that if we find a secret that is no longer referenced by any MR to remove the Label `ReferencedBy` if present?
Something like this?
``` go
if !secretIsReferenced && metav1.HasLabel(secret.ObjectMeta, resourcesv1alpha1.ReferencedBy) {
	patch := client.MergeFromWithOptions(secret.DeepCopy(), client.MergeFromWithOptimisticLock{})
	delete(secret.Labels, resourcesv1alpha1.ReferencedBy)
	if err := r.SourceClient.Patch(ctx, secret, patch); err != nil {
		return reconcile.Result{}, fmt.Errorf("failed to remove the %s label to secret: %w", resourcesv1alpha1.ReferencedBy, err)
	}
}
```